app/services/subject_mapper_service.rb

```diff
@@ -92,6 +92,8 @@ class SubjectMapperService
   }.freeze
 
   def self.get_subject_list(course_title, ucas_subjects)
+    return [] if course_title.nil?
```

Comment:
```suggestion
    return [] if course_title.blank?
```

Reply:
I think I was assuming that we only want to guard against uninitialised values, i.e. for a new course, and behave as before for a title that has been set to empty string. Also, it seemed less intrusive to only change the behaviour for nil. No?

Reply:
hmmm... coming to think of it a course should have a title regardless either from 
-  user input (for adding new courses) or database (for existing courses)
- or calculated from the subjects list (for adding new courses).

I'm not to sure how this `self.get_subject_list` needs to know that if `title is not set` then `return no subjects`.

maybe scope creep...

original reason for the subject mapper needing the title is because a `English` course may not have a subject associated with it.

Reply:
This guard was added purely to allow the API to return a "new" version of the course. We to be able to:

1. not set any attributes, but validate those that are set so that we can start to return meaningful feedback to the user in the "create course" wizard
2. return some of the metadata that's required by the "create course" wizard to populate the steps required to create a course

So we need a way to allow a course that has no title set (i.e. `nil`) to still have this method called on it without blowing up. That could happen here, as in this PR, or in `Course#dfe_subjects` if that's better, I'm not too fussed tbh.  Would that be a better place for it?

Reply:
Just off the back of the comment below for `with_qualification_spec.rb`, I want to re-iterate the same here: It's not valid to have a course have no title, but that should be caught by a validation since until we persist to the db, we want to allow invalid course objects to be instantiated so that we can help out the wizard.